### PR TITLE
Drop the [...] link on index pages

### PR DESCRIPTION
**Pull request by kevmoo**
Fixes https://github.com/dart-lang/dartdoc/issues/2928

### CODE REVIEW

**Comment by kevmoo** [comment]
Feel free to squash/merge and delete the branch if you're happy

**Comment by srawlins** [comment]
What does a chopped off doc comment look like with out `[...]`?

**Comment by kevmoo** [comment]
It's almost ALWAYS a full sentence. See https://api.dart.dev/stable/2.16.0/index.html

I'd love to see an example where it isn't....

**Comment by parlough** [comment]
I thought it just shows the first line then shows this if there are more lines?

https://github.com/dart-lang/dartdoc/blob/master/lib/src/markdown_processor.dart#L371

**Comment by kevmoo** [comment]
> I thought it just shows the first block (which I believe can be multiple source lines?) then showed this if there are more blocks?
> 
> https://github.com/dart-lang/dartdoc/blob/master/lib/src/markdown_processor.dart#L371

Godo catch, @parlough – did some more cleanup!

File: lib/src/markdown_processor.dart [hunk 1]
```diff
@@ -357,25 +357,19 @@ class MarkdownDocument extends md.Document {
             linkResolver: linkResolver,
             imageLinkResolver: imageLinkResolver);
 
-  /// Parses markdown text, collecting the first [md.Node] or all of them
-  /// if [processFullText] is `true`. If more than one node is present,
-  /// then [DocumentationParseResult.hasExtendedDocs] will be set to `true`.
-  DocumentationParseResult parseMarkdownText(
-      String text, bool processFullText) {
-    var hasExtendedContent = false;
```
Comment (parlough) [inline lib/src/markdown_processor.dart:365]: Can you add the docs back?

```dart
/// Parses markdown text, collecting the first [md.Node] 
/// or all of them if [processFullText] is `true`.
```

File: lib/src/markdown_processor.dart [hunk 2]
```diff
@@ -357,25 +357,19 @@ class MarkdownDocument extends md.Document {
...
+  List<md.Node> parseMarkdownText(String text, bool processFullText) {
     var lines = LineSplitter.split(text).toList();
     md.Node? firstNode;
     var nodes = <md.Node>[];
     for (var node in _IterableBlockParser(lines, this).parseLinesGenerator()) {
       if (firstNode != null) {
-        hasExtendedContent = true;
         if (!processFullText) break;
       }
```
Comment (parlough) [inline lib/src/markdown_processor.dart:367]:
```dart
if (firstNode != null & !processFullText) {
  break;
}

**Review by parlough — COMMENTED** [review]
I love the extra cleanup! I think you'll need to rerun `dart run grinder build` again.

File: lib/src/markdown_processor.dart (hunk 1 above)
Comment (kevmoo) [reply lib/src/markdown_processor.dart:365]: ack

File: lib/src/markdown_processor.dart (hunk 2 above)
Comment (kevmoo) [reply lib/src/markdown_processor.dart:367]: ack

**Comment by kevmoo** [comment]
Thoughts @srawlins ?

**Comment by srawlins** [comment]
Ok it does look like its always a full sentence because its the first line. What is the indication that there is more doc comment beyond what is shown?

**Comment by kevmoo** [comment]
> Ok it does look like its always a full sentence because its the first line. What is the indication that there is more doc comment beyond what is shown?

I don't think it's necessary. If folks want to know more, they will click on the entry. I don't see this affordance in any other API docs I've reviewed (and I've looked at pretty much all of them 😄 )

**Comment by parlough** [comment]
I took a look at the SDK docs without the [...] and I have to say the pages look less cluttered and my attention is drawn to each member name instead. I think clicking on the member name for more information is pretty intuitive as it already presents as a link and in most cases the first line is sufficient anyway.

**Comment by srawlins** [comment]
Yeah I'd definitely prefer a review from a UX expert.

**Comment by kevmoo** [comment]
> Yeah I'd definitely prefer a review from a UX expert.

Understood, @srawlins . I hope to engage UX here for bigger items. In the meantime, look at Microsoft, Android, Apple, etc – no one uses the `[...]` anywhere. For places where we're clearly being unnecessarily different than other thought leaders here, we should just get in line – unless there's a good reason not to.

**Comment by srawlins** [comment]
I agree.

Can you show examples which parallel our style of printing a 1-line summary of a doc, and a complete doc on a different page? I couldn't find any examples in Android, but I see that Rust uses the words, "Read more": https://doc.rust-lang.org/core/primitive.array.html#method.borrow_mut

**Comment by kevmoo** [comment]
Microsoft: https://docs.microsoft.com/en-us/dotnet/api/system?view=net-6.0

The output has a "read more" header section, which seems gratuitous.

Android:
https://developer.android.com/reference/androidx/test/espresso/accessibility/package-summary

Shows the first line of a description. Clicking on AccessibilityChecks
<https://developer.android.com/reference/androidx/test/espresso/accessibility/AccessibilityChecks>
takes
you to the full description.

**Review by srawlins — APPROVED** [review]
OK that's good enough for me. Mind re-basing and regenerating those files, Kevin?

**Comment by kevmoo** [comment]
@srawlins – note version change here, too!

**Pull request merged**